core/build.gradle

```diff
@@ -54,6 +122,10 @@ dependencies {
     compile ([group: 'org.apache.httpcomponents', name: 'httpclient', version: '4.5.3'])
     compile ([group: 'com.google.guava', name: 'guava', version: '20.0'])
     compile ([group: 'com.google.code.gson', name: 'gson', version: '2.8.2'])
+    compile ([group: 'com.google.protobuf', name:'protobuf-java', version:'3.6.1'])
+    compile ([group: 'io.grpc', name: 'grpc-stub', version: '1.14.0'])
+    compile ([group: 'io.grpc', name: 'grpc-protobuf', version: '1.14.0'])
+    runtime ([group: 'io.grpc', name: 'grpc-netty-shaded', version: '1.14.0'])
```

Comment:
'runtime' is different than 'provided' (don't forget we added provided; runtime is part of gradle).

A 'runtime' class is not needed for compile, but it is needed at runtime. Shadow knows about this and includes it in the repackaging. Generally, you won't be referencing a 'runtime' class directly.

A 'provided' class is included in the compile classpath, but it will not be repackaged. The user must provide it at runtime. Provided classes are generally referenced in source.

Classes in 'grpc-netty-shaded' are loaded via reflection by grpc at runtime.
In contrast, 'servlet-api' is referenced in our web code, but we use the application server's copy instead of repackaging.

Reply:
ok thanks